### PR TITLE
Bani-2024-01-03

### PR DESCRIPTION
**Pull request by Bani2791**
1. Carrier section
Carrier - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

2. Alert: Pushover Notifications
Alerts - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

3. Customer Portal: Data Management
Data Management - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

4. Customer Portal: Status
Status - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

5. Jobs
Jobs - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

6. Change Log
Change Log - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

7. Date/Time Restrict
https://bani-2024-1-3--connexcs-docs.netlify.app/customer/customer/#create-customer

With this feature, you can refrain your customers from calling at specified time slots.

How to use?

Login to your Control Panel.
Navigate to Management  Customer  Customer [Name]  Day/Time Restrict.
Choose the Timezone from the drop-down.
Drag the time slider between the time-slots you wish to aloow your customers to dial.
Click Save.

Example

In the image below you can see the time slider starts from 12:00 AM till 8:00 AM, this means your customers are only allowed to dial between 12:00 AM and 8:00 AM. After 8:00 AM your customers won't be able to call. The rest of the slots (grey color) are restricted slots.

8. Refer ConnnexCS > Customer and Carrier
8.1 Customer
https://bani-2024-1-3--connexcs-docs.netlify.app/customer/customer/

Refer ConnexCS: This feature allows you to send simple ConnexCS Referrals to you customers.
You can send referrals to multiple Customers.
You can select the Templates from the drop-down list.
Enter the Subject.
Write the Body of the referral.
Click on Send.

8.2 Carrier
https://bani-2024-1-3--connexcs-docs.netlify.app/carrier/
+ **Send**: ConnexCS includes Refer ConnexCS for your customers.

1. Select several customers using the tick-box selector on the left side of each row. (Customers aren't visible to each other in the sent message.)
2. Click on "Send" at the top right then choose Refer ConnexCS.
3. Refer ConnexCS: This feature allows you to send simple ConnexCS Referrals to you customers.
    + You can send referrals to multiple **Customers**.
    + You can select the **Template** from the drop-down list.
    + Enter the **Subject**.
    + Write the Body of the referral.
    + Click on `Send`.
  

9. Alerts: Notice Level

https://bani-2024-1-3--connexcs-docs.netlify.app/customer/alerts/

Notice Level: You can assign alert levels like Warning or Error for classifying and communicating the severity of events or conditions that require attention or action.

10. Dashboard Graphs
Dashboard: Graph Tabs¶
With this new feature, you can view parameters like Channels, CPS, Customer/Provider Spend, Profit and Registration Count.

You can view the Configuration of the displayed graph. The feature gives you the flexibility for Resizing the graph, Refreshing it and Download it as well.

Channels
CPS
Customer/Provider Spend
Profit
Registration Count

Dashboard - ConnexCS Documentation (bani-2024-1-3--connexcs-docs.netlify.app)

11. Download CDR report in Main

* **Download CDR Report**: You now have the option to **Download** the **CDR** report in the **.csv** format. You can download the CDR report as a **Daily**, **Weekly** or **Monthly** time-frame. You just need to click on the download icon Alt text.

https://bani-2024-1-3--connexcs-docs.netlify.app/customer/main/#summary

12. Add ConneXML to DID
https://bani-2024-1-3--connexcs-docs.netlify.app/customer/did/#add-connexml-tab-to-did

### CODE REVIEW

File: docs/carrier.md [hunk 1]
```diff
@@ -215,6 +215,34 @@ To ensure compatibility, set the switch manufacturer to `bandwidth.com`.
 [carriers-main]: /carrier/img/carriers-main.png "Carriers"
 [carrier]: /carrier/img/carrier.png "Carrier Main"
 
+## Headers from Customers to Providers
+
+When the Carrier requires information on the headers sent from the UAC, you can use below discussed process:
```
Comment (AdamNawaz-connex) [inline docs/carrier.md:220]: Customize the handling of X-Headers between the User Agent Client (UAC) and the carrier. By default, all UAC-originating X-Headers are removed for standardization, but this configuration allows selective retention of specific headers crucial for carrier communication.

File: docs/carrier.md [hunk 2]
```diff
@@ -215,6 +215,34 @@ To ensure compatibility, set the switch manufacturer to `bandwidth.com`.
 [carriers-main]: /carrier/img/carriers-main.png "Carriers"
 [carrier]: /carrier/img/carrier.png "Carrier Main"
 
+## Headers from Customers to Providers
```
Comment (AdamNawaz-connex) [inline docs/carrier.md:218]: Add which headers as this is for X headers only

File: docs/carrier.md [hunk 3]
```diff
@@ -215,6 +215,34 @@ To ensure compatibility, set the switch manufacturer to `bandwidth.com`.
...
+
+When the Carrier requires information on the headers sent from the UAC, you can use below discussed process:
+
+1. Login to your account.
+2. Navigate to Management :material-menu-right: Customer :material-menu-right: Carrier :material-menu-right: blue `+` button :material-menu-right: Config :material-menu-right: Vars<sup>TOML</sup>.
+3. We consider an example here: **Example of how to retain X-ng199 but remove all other X-headers**
```
Comment (AdamNawaz-connex) [inline docs/carrier.md:224]: consider an example here

File: docs/customer-portal/cp-data-management.md [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+# Database
+
+The **Database** allows you to create custom persistent data storage. A datastore can be specific to a single customer or global (used with any customer).
+
+You can use **Database** for activities like checking list of numbers, caching external lookups, or creating stateful applications that provide users a set number of free minutes per month.
+
+## Steps to create a Database
+
+1. Login to your **Control Panel**.
+2. Click on **IDE** :material menu-right: +.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-data-management.md:10]: left click

Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-data-management.md:10]: :material menu-right: 
not correct.

File: docs/customer-portal/cp-data-management.md [hunk 5]
```diff
@@ -1,3 +1,30 @@
...
+5.Select the **Customer** you want to create the database for under the **Basic** Tab.
+
+6.Select from the **Dataset Type** which are the preset data storage options. The two options are:
+
++ **Key / Value** - Contains a basic Key / Value Storage.
++ **Lead Set** - Contains the most common fields when using a datastore for customer / campaign data which you can use with dialers.
+
+<img src="/developers/img/db2.png" width="300">
+
+7.Next to the **Basic** Tab is the **SQL Table Schema**. This allows you to add additional fields apart from the **Leadset**. You can also select it's **Type** like integer, string, double, float etc.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-data-management.md:26]: Add additional fields  to leadset, change order, remove a field

File: docs/customer-portal/cp-data-management.md [hunk 6]
```diff
@@ -1,3 +1,30 @@
...
+<img src= "/customer-portal/img/app_database.png" width="400">
+
+4.Under the **Basic** Tab Write the **Name** of the database you wish to create.
+
+5.Select the **Customer** you want to create the database for under the **Basic** Tab.
+
+6.Select from the **Dataset Type** which are the preset data storage options. The two options are:
+
++ **Key / Value** - Contains a basic Key / Value Storage.
++ **Lead Set** - Contains the most common fields when using a datastore for customer / campaign data which you can use with dialers.
```
Comment (AdamNawaz-connex) [inline docs/customer-portal/cp-data-management.md:22]: Leadset

File: docs/customer/alerts.md [hunk 7]
```diff
@@ -70,6 +70,52 @@ You can test the created alert under a specific customer.
...
+7.You will see the following message on the top **Your message has been queued for delivery**.
+
+<img src= "/customer/img/pushover2.png ">
+
+8.Received the message on https://client.pushover.net/ .
+
+<img src= "/customer/img/pushover3.png ">
+
+9.You now have Push message on the device.
+
```
Comment (AdamNawaz-connex) [inline docs/customer/alerts.md:104]: steps 4 to 9 seem surplus.

File: docs/setup/information/jobs.md [hunk 8]
```diff
@@ -1,14 +1,30 @@
...
+4. **Delayed**: When the task has been delayed.
+5. **Wait**: The task has been paused.
+
+## Types of Tasks
 
-### Types of Status 
+1. **Card**: You can check the status of the Rate Card, when you build it.
+2. **CDR**: You can check the status of any re-rate.
+3. **Delayed**: You can check the status of any task you've delayed.
+4. **ORC**: You can the status of any server which has been added or restarted.
```
Comment (AdamNawaz-connex) [inline docs/setup/information/jobs.md:20]: check

File: docs/setup/information/jobs.md [hunk 9]
```diff
@@ -1,14 +1,30 @@
...
+5. **Send**: You can check the status of the e-mails you send. You can even re-try to send the e-mail along with failed reasons.
+6. **ScriptForge**: You can check the status of the ScriptForge running in the background.
+7. **Switch**: You can check the status of any changes are made to the OpenSIPS.
```
Comment (AdamNawaz-connex) [inline docs/setup/information/jobs.md:23]: check

File: docs/setup/information/jobs.md [hunk 10]
```diff
@@ -1,14 +1,30 @@
...
+5. **Wait**: The task has been paused.
+
+## Types of Tasks
 
-### Types of Status 
+1. **Card**: You can check the status of the Rate Card, when you build it.
+2. **CDR**: You can check the status of any re-rate.
+3. **Delayed**: You can check the status of any task you've delayed.
+4. **ORC**: You can the status of any server which has been added or restarted.
+5. **Send**: You can check the status of the e-mails you send. You can even re-try to send the e-mail along with failed reasons.
```
Comment (AdamNawaz-connex) [inline docs/setup/information/jobs.md:21]: check

File: docs/carrier.md (hunk 1 above)
Comment (Bani2791) [reply docs/carrier.md:220]: added in the introduction

File: docs/customer-portal/cp-data-management.md (hunk 6 above)
Comment (Bani2791) [reply docs/customer-portal/cp-data-management.md:22]: corrected

File: docs/customer-portal/cp-data-management.md [hunk 11]
```diff
@@ -1,3 +1,30 @@
...
+
+<img src="/developers/img/db2.png" width="300">
+
+7.Next to the **Basic** Tab is the **SQL Table Schema**. This allows you to add additional fields apart from the **Leadset**. You can also select it's **Type** like integer, string, double, float etc.
+
+<img src="/developers/img/db3.png" width="400">
+
+8.Click on `Save` to finally create the database.
```
[the remark this answers is not in the document]
Comment (Bani2791) [reply docs/customer-portal/cp-data-management.md:30]: removed

File: docs/setup/information/jobs.md (hunk 8 above)
Comment (Bani2791) [reply docs/setup/information/jobs.md:20]: added check in the ORC.

File: docs/setup/information/jobs.md (hunk 9 above)
Comment (Bani2791) [reply docs/setup/information/jobs.md:23]: added check in the ORC.

File: docs/carrier.md (hunk 2 above)
Comment (Bani2791) [reply docs/carrier.md:218]: done

File: docs/carrier.md (hunk 3 above)
Comment (Bani2791) [reply docs/carrier.md:226]: done

File: docs/customer-portal/cp-data-management.md (hunk 5 above)
Comment (Bani2791) [reply docs/customer-portal/cp-data-management.md:26]: done

**Review by AdamNawaz-connex — APPROVED** [review]

**Pull request merged**